includes/class-wc-stripe.php

```diff
@@ -857,4 +860,39 @@ public function initialize_status_page() {
 		$wcstripe_status = new WC_Stripe_Status( self::get_main_stripe_gateway(), $this->account );
 		$wcstripe_status->init_hooks();
 	}
+
+	/**
+	 * Maybe deactivate Affirm or Klarna payment methods if other official plugins are active.
+	 *
+	 * @return void
+	 */
+	public function maybe_deactivate_bnpls() {
+		$has_affirm_plugin_active = WC_Stripe_Helper::has_gateway_plugin_active( WC_Stripe_Helper::OFFICIAL_PLUGIN_ID_AFFIRM );
+		$has_klarna_plugin_active = WC_Stripe_Helper::has_gateway_plugin_active( WC_Stripe_Helper::OFFICIAL_PLUGIN_ID_KLARNA );
+		if ( ! $has_affirm_plugin_active && ! $has_klarna_plugin_active ) {
+			return;
+		}
+
+		$gateway = $this->get_main_stripe_gateway();
+
+		$enabled_payment_methods       = $gateway->get_upe_enabled_payment_method_ids();
+		$payment_method_ids_to_disable = [];
+		if ( in_array( WC_Stripe_Payment_Methods::AFFIRM, $enabled_payment_methods, true ) && $has_affirm_plugin_active ) {
+			$payment_method_ids_to_disable[] = WC_Stripe_Payment_Methods::AFFIRM;
+		}
+		if ( in_array( WC_Stripe_Payment_Methods::KLARNA, $enabled_payment_methods, true ) && $has_klarna_plugin_active ) {
+			$payment_method_ids_to_disable[] = WC_Stripe_Payment_Methods::KLARNA;
+		}
```

Comment:
Non-blocking nit: I would prefer to check the pre-computed value first in each case.
```suggestion
		if ( $has_affirm_plugin_active && in_array( WC_Stripe_Payment_Methods::AFFIRM, $enabled_payment_methods, true ) ) {
			$payment_method_ids_to_disable[] = WC_Stripe_Payment_Methods::AFFIRM;
		}
		if ( $has_klarna_plugin_active && in_array( WC_Stripe_Payment_Methods::KLARNA, $enabled_payment_methods, true ) ) {
			$payment_method_ids_to_disable[] = WC_Stripe_Payment_Methods::KLARNA;
		}
```